crates/iota-rest-kv/src/server.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module includes helper wrappers for building and starting a REST API
+//! server.
+
+use std::{net::SocketAddr, sync::Arc};
+
+use anyhow::Result;
+use axum::{Router, response::IntoResponse, routing::get};
+use tokio_util::sync::CancellationToken;
+
+use crate::{
+    RestApiConfig,
+    errors::ApiError,
+    kv_store_client::KvStoreClient,
+    routes::{health, kv_store},
+};
+
+/// A wrapper which builds the components needed for the REST API server and
+/// provides a simple way to start it.
+pub struct Server {
+    router: Router,
+    server_address: SocketAddr,
+    token: CancellationToken,
+}
+
+impl Server {
+    /// Create a new Server instance.
+    ///
+    /// Based on the config, it instantiates the [`KvStoreClient`] and
+    /// constructs the [`Router`].
+    pub async fn new(config: RestApiConfig, token: CancellationToken) -> Result<Self> {
+        let kv_store_client = KvStoreClient::new(config.kv_store_config).await?;
+
+        let shared_state = Arc::new(kv_store_client);
+
+        let router = Router::new()
+            .route("/health", get(health::health))
+            .route("/:key/:item_type", get(kv_store::data_as_bytes))
```

Comment:
Wouldn't it make more sense to reverse the order into `/:item_type/:key` so that the `item_type` defines the namespace for a possible set of underlying requests. What would we do for example if we wanted to add more methods for given `item_type`.

I see that the current route is expected in `iota_storage`, but doesn't seem right. What do you think?

Reply:
yes, I agree with you 100%, It didn't seem right for me also, I'll happily apply thins change

Reply:
[fixed](https://github.com/iotaledger/iota/pull/5296/commits/e4605179fb0a57b8814a097c4bc1b9436d69307f)